test/unit/uk/gov/hmrc/individualsbenefitsandcreditsapi/audit/AuditHelperSpec.scala

```diff
@@ -137,70 +127,37 @@ class AuditHelperSpec extends UnitSpec with MockitoSugar {
-          |  "response": "Something went wrong",
-          |  "method": "GET",
-          |  "deviceID": "-",
-          |  "ipAddress": "-",
-          |  "referrer": "-",
-          |  "Authorization": "-",
-          |  "input": "Request to /",
-          |  "userAgentString": "-"
-          |}
-          |""".stripMargin)
-
-      val capturedEvent = captor.getValue
-      capturedEvent.asInstanceOf[ExtendedDataEvent].auditSource shouldEqual "individuals-benefits-and-credits-api"
-      capturedEvent.asInstanceOf[ExtendedDataEvent].auditType shouldEqual "ApiFailureEvent"
-      capturedEvent.asInstanceOf[ExtendedDataEvent].detail shouldBe result
+      verify(auditConnector, times(1)).sendExplicitAudit(eqTo("ApiFailureEvent"),
+        captor.capture())(any(), any(), any())
+
+      val capturedEvent = captor.getValue.asInstanceOf[ApiFailureResponseEventModel]
+      capturedEvent.matchId shouldEqual matchId
+      capturedEvent.correlationId shouldEqual Some(correlationId)
+      capturedEvent.requestUrl shouldEqual endpoint
+      capturedEvent.response shouldEqual msg
 
     }
 
     "auditIfApiResponse" in {
 
       Mockito.reset(auditConnector)
 
-      val captor = ArgumentCaptor.forClass(classOf[ExtendedDataEvent])
-
-      auditHelper.auditIfApiResponse(correlationId, scopes, matchId, request, ifUrl, response)
-
-      verify(auditConnector, times(1)).sendExtendedEvent(captor.capture())(any(), any())
-
-      val result = Json.parse(
-        """
-          |{
-          |  "apiVersion": "2.0",
-          |  "matchId": "80a6bb14-d888-436e-a541-4000674c60aa",
-          |  "correlationId": "test",
-          |  "scopes": "test",
-          |  "requestUrl": "host/individuals/benefits-and-credits/child-tax-credit/nino/CS700100A?startDate=2019-01-01&endDate=2020-01-01&fields=some(vals(val1),val2)",
-          |  "response": "[\"some\",\"json\"]",
-          |  "method": "GET",
-          |  "deviceID": "-",
-          |  "ipAddress": "-",
-          |  "referrer": "-",
-          |  "Authorization": "-",
-          |  "input": "Request to /",
-          |  "userAgentString": "-"
-          |}
-          |""".stripMargin)
-
-      val capturedEvent = captor.getValue
-      capturedEvent.asInstanceOf[ExtendedDataEvent].auditSource shouldEqual "individuals-benefits-and-credits-api"
-      capturedEvent.asInstanceOf[ExtendedDataEvent].auditType shouldEqual "IfApiResponseEvent"
-      capturedEvent.asInstanceOf[ExtendedDataEvent].detail shouldBe result
+      val captor = ArgumentCaptor.forClass(classOf[IfApiResponseEventModel])
+
+      auditHelper.auditIfApiResponse(correlationId, matchId, request, ifUrl, ifResponse)
+
+      verify(auditConnector, times(1)).sendExplicitAudit(eqTo("IfApiResponseEvent"),
+        captor.capture())(any(), any(), any())
+
+      val capturedEvent = captor.getValue.asInstanceOf[IfApiResponseEventModel]
+      capturedEvent.asInstanceOf[IfApiResponseEventModel].matchId shouldEqual matchId
```

Comment:
capturedEvent is already an IfApiResponseEventModel from line above

Reply:
Good spot